File: iosApp/iosApp/views/screens/settings/SettingsScreenEvents.swift
import SwiftUI
import Shared


class SettingsEventHandler {
    private let events: Events

    init(events: Events) {
        self.events = events
    }
 
    func onSettingAction(_ action: SettingAction) {
        if let action = action as? SettingAction.SetFavoriteSwipe {
            events.setFavoriteSwipeEnabled(enabled: action.enabled)
        } else if let action = action as? SettingAction.SetThemeMode {
            events.saveThemeMode(name: action.value)
        } else if action is SettingAction.SyncData {
            events.syncDataFromSettings()
        }
    }
}

File: composeApp/src/androidMain/kotlin/com/exploramus/app/composables/screens/settings/SettingsScreenEvents.kt
package com.exploramus.app.composables.screens.settings

import com.exploramus.shared.viewmodel.core.Events
import com.exploramus.shared.viewmodel.screens.settings.builder.SettingAction
import com.exploramus.shared.viewmodel.screens.settings.saveThemeMode
import com.exploramus.shared.viewmodel.screens.settings.setFavoriteSwipeEnabled
import com.exploramus.shared.viewmodel.screens.settings.syncDataFromSettings

sealed class SettingsUiEvent

class SettingsEventHandler(
    val events: Events
) {
    fun onEvent(event: SettingsUiEvent) {}

    fun onSettingAction(action: SettingAction) {
        when (action) {
            is SettingAction.SetFavoriteSwipe -> {
                events.setFavoriteSwipeEnabled(action.enabled)
            }

            is SettingAction.SetThemeMode -> {
                events.saveThemeMode(name = action.value)
            }

            SettingAction.SyncData -> {
                events.syncDataFromSettings()
            }
        }
    }
}

